### PR TITLE
[Doc] Add a section for synchronization primitives in Ray

### PR DESCRIPTION
**Pull request by simon-mo**
## Why are these changes needed?
Document several things:
- FileLock
- SignalActor
- Queue

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: doc/source/package-ref.rst [hunk 1]
```diff
@@ -135,6 +135,15 @@ Experimental APIs
 .. automodule:: ray.experimental
    :members:
 
+.. _ray-queue-ref:
+
+ray.experimental.queue
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. autoclass:: ray.experimental.queue.Queue
+   :members:
```
Comment (richardliaw) [inline doc/source/package-ref.rst:144]: @ericl is this OK? seems to be useful for those implementing synchronization (see above documentation)

**Review by richardliaw — COMMENTED** [review]
Looks good to me but will leave it to Eric to approve

File: doc/source/package-ref.rst (hunk 1 above)
Comment (ericl) [reply doc/source/package-ref.rst:144]: Hmm I would actually advocate removing the experimental API section entirely. I think users can just read the source code if they really want to use experimental features.

Comment (ericl) [reply doc/source/package-ref.rst:144]: (and we should be aggressive about promoting experimental to mainline).

**Review by ericl — APPROVED** [review]

File: doc/source/package-ref.rst (hunk 1 above)
Comment (richardliaw) [reply doc/source/package-ref.rst]: ok removed

Comment (simon-mo) [reply doc/source/package-ref.rst]: kk that should be a separate PR then.

**Pull request merged**